api/src/main/java/org/apache/iceberg/metrics/MetricsContext.java

```diff
@@ -126,15 +126,14 @@ public Timer timer(String name, TimeUnit unit) {
 
       @Override
       public <T extends Number> Counter<T> counter(String name, Class<T> type, Unit unit) {
-        return new Counter<T>() {
-          @Override
-          public void increment() {
-          }
-
-          @Override
-          public void increment(T amount) {
-          }
-        };
+        if (Integer.class.equals(type)) {
+          return (Counter<T>) IntCounter.NOOP;
+        }
+
+        if (Long.class.equals(type)) {
+          return (Counter<T>) LongCounter.NOOP;
+        }
+        throw new IllegalArgumentException(String.format("Counter for type %s is not supported", type.getName()));
```

Comment:
Style: there should be a newline between a control flow block and the follow statement.

Reply:
fixed